### PR TITLE
rendering meal text

### PR DESCRIPTION
**Pull request by BinaHovav**
image
Hi I added rendering the text of each meal

### CODE REVIEW

File: src/components/Meals.tsx [hunk 1]
```diff
@@ -1,66 +1,65 @@
...
-  handleBreakfastText: (text: string, category: MealCategories) => void;
+  handleMealText: (text: MealText, category: MealCategories) => void;
 };
 
-const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleBreakfastText }) => {
-  const [textMeal, setMealText] = useState('');
-  const [breakfastText, setBreakfastText] = useState('');
-  const [lunchText, setLunchText] = useState('');
-  const [dinnerText, setDinnerText] = useState('');
+type MealText = string;
```
Comment (immanuelAjz) [inline src/components/Meals.tsx]: this should be in the model file

File: src/components/Meals.tsx [hunk 2]
```diff
@@ -1,66 +1,65 @@
 import React, { useState } from 'react';
-import { View, Text, TextInput, TouchableOpacity, Button } from 'react-native';
+import { View, Text, TextInput, TouchableOpacity } from 'react-native';
 import styles from '../styles';
 import { MealCategories } from '../models/meals.model';
 
 type MealsComponentProps = {
```
Comment (immanuelAjz) [inline src/components/Meals.tsx]: this should be in the model file

File: src/components/Meals.tsx [hunk 3]
```diff
@@ -1,66 +1,65 @@
...
+  });
+
+  const onMealEnter = (text: MealText, category: MealCategories) => {
+    setMealText((prev) => ({
+      ...prev,
+      [category]: text,
+    }));
   };
 
   const getCategoryInput = (category: MealCategories) => {
```
Comment (immanuelAjz) [inline src/components/Meals.tsx]: not sure u need a function for this, but if yes then u can shorten it like this if u want
```suggestion
  const getCategoryInput = (category: MealCategories) =>  mealText[category];
```

File: src/components/Meals.tsx [hunk 4]
```diff
@@ -1,66 +1,65 @@
...
-        setLunchText(text);
-        break;
-      case MealCategories.DINNER:
-        setDinnerText(text);
-        break;
-      default:
-        break;
-    }
+const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleMealText }) => {
+  const [mealText, setMealText] = useState<Record<MealCategories, MealText>>({
```
Comment (immanuelAjz) [inline src/components/Meals.tsx]: why do we need the word `Text` in the name? isn't that a little too abstract? (same for `showEnteredText`)
what is this exactly a state of?

File: src/components/Meals.tsx (hunk 1 above)
Comment (BinaHovav) [reply src/components/Meals.tsx]: fixed

File: src/components/Meals.tsx (hunk 2 above)
Comment (BinaHovav) [reply src/components/Meals.tsx]: fixed

File: src/components/Meals.tsx (hunk 3 above)
Comment (BinaHovav) [reply src/components/Meals.tsx]: ya i got rid of the function

File: src/components/Meals.tsx (hunk 4 above)
Comment (BinaHovav) [reply src/components/Meals.tsx]: it refers to the text the user wrote, i can change it to input

File: src/components/Meals.tsx (hunk 2 above)
Comment (immanuelAjz) [reply src/components/Meals.tsx]: Nope. Still here

File: src/components/Meals.tsx (hunk 4 above)
Comment (immanuelAjz) [reply src/components/Meals.tsx]: Sounds better

File: src/components/Meals.tsx (hunk 2 above)
Comment (BinaHovav) [reply src/components/Meals.tsx]: now it's not

**Review by immanuelAjz — APPROVED** [review]
Well done!

**Pull request merged**